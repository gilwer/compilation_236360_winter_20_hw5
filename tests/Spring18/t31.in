int zero() { return 0; }
byte zeroB() { return 0b; }
int printAndRet() {
    print("called printAndRet");
    return 1;
}
int foo(
    int a0,
    int a1,
    int a2,
    int a3,
    int a4,
    int a5,
    int a6,
    int a7,
    int a8,
    int a9
) {
    print("a6=");
    printi(a6);
    print("a9=");
    printi(a9);
    
	return 0;
}

int bar(int x, int y)
{
    return y-x;
}

int fu(int x) {
    return 10-x;
}


void main() {
    printi(zero());
    
    printi(zeroB());
    
    printi(zero() + (1 + (2 + (3 + (4 + (5 + (6 + (7 + (8 + (9 + (10 + (11 + (12 + (13 + zeroB()))))))))))))));
    printAndRet();
    printi(zero() + zero() + zeroB());
    printi(printAndRet());
	foo(0,1,2,3,4,5,6,7,8,9);
    foo(0,1+bar(4,19),2*19,3,4,5,6+bar(2,3)/fu(0),7,8,9-fu(20));
}