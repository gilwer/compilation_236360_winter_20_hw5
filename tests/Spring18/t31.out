0
0
91
called printAndRet
0
called printAndRet
1
a6=
6
a9=
9
a6=
6
a9=
19
